src/RemediationRoleSaver.cpp

```diff
@@ -24,24 +24,53 @@
 
 #include <QFile>
 
+#include "RemediationRoleSaver.h"
+#include "DiagnosticsDialog.h"
+
 extern "C"
 {
 #include <oscap_error.h>
 #include <xccdf_benchmark.h>
 #include <xccdf_policy.h>
 #include <xccdf_session.h>
+#ifdef SCAP_WORKBENCH_USE_LIBRARY_FOR_RESULT_BASED_REMEDIATION_ROLES_GENERATION
+    // vvv This include is used only for library-based generation of result-base remediation roles
+    // vvv and it requires (relatively recent) openscap 1.2.16
+#include <ds_rds_session.h>
+#endif
 }
 
-#include "RemediationRoleSaver.h"
+#include "TemporaryDir.h"
+
+
+const QString bashSaveMessage = QObject::tr("Save remediation role as a bash script");
+const QString bashFiletypeExtension = "sh";
+const QString bashFiletypeTemplate = QObject::tr("bash script (*.%1)");
+const QString bashFixTemplate = QString("sh");
+
+const QString ansibleSaveMessage = QObject::tr("Save remediation role as an ansible playbook");
+const QString ansibleFiletypeExtension = "yml";
+const QString ansibleFiletypeTemplate = QObject::tr("ansible playbook (*.%1)");
+const QString ansibleFixType = QString("ansible");
+
+const QString puppetSaveMessage = QObject::tr("Save remediation role as a puppet manifest");
+const QString puppetFiletypeExtension = "pp";
+const QString puppetFiletypeTemplate = QObject::tr("puppet manifest (*.%1)");
+const QString puppetFixType = QString("puppet");
+
+
+RemediationSaverBase::RemediationSaverBase(QWidget* parentWindow,
+                const QString& saveMessage, const QString& filetypeExtension, const QString& filetypeTemplate, const QString& fixType):
+    mParentWindow(parentWindow), mDiagnostics(globalDiagnosticsDialog), mSaveMessage(saveMessage), mFiletypeExtension(filetypeExtension), mFiletypeTemplate(filetypeTemplate),
```

Comment:
why do you create another member and set it to globalDiagnosticsDialog?

Reply:
A part of the class is isolated from the fact that there is a global variable out there. Getting rid of it will be slightly easier thanks to this separation.